token-syncer/README.md

```diff
@@ -0,0 +1,65 @@
+The token-syncer is a command-line application designed specifically for syncing tokens across Waiter clusters.
+This application can be set up as a cronjob to run at intervals to sync tokens on independently running Waiter clusters on different datacenters.
+The syncing allows supporting datacenter failures as we can failover to another datacenter either with minimal downtime and customer impact.
+All that is needed in the datacenter failure scenario is to point the DNS records to an active datacenter running a Waiter instance.
+As requests come in to the new Waiter instance, Waiter will spin up service instances on the new datacenter and process the requests.
+
+
+# Usage
+
+The token-syncer is a command-line application written in Clojure.
+It can be run as follows:
+
+```bash
+$ lein run --help
+<displays usage instructions>
+```
+
+```bash
+$ lein run --cluster-urls "http://cluster-1.localtest.me:9091;http://cluster-2.localtest.me:9093"
+...
+Syncing tokens on clusters: \#{http://cluster-1.localtest.me:9091 http://cluster-2.localtest.me:9093}
+...
+Exiting.
+```
+
+Alternatively, it can also be run after building the uberjar (see below):
+```bash
+$ java -jar /path/to/token-syncer.jar --cluster-urls "http://cluster-1.localtest.me:9091;http://cluster-2.localtest.me:9093"
+...
+Syncing tokens on clusters: \#{http://cluster-1.localtest.me:9091 http://cluster-2.localtest.me:9093}
+...
+Exiting.
+```
+
+# Implementation
+
+The token syncer communicates with Waiter clusters on their respective `/token` endpoints to retrieve and manage individual tokens.
+When requested to sync tokens across multiple clusters, it first loads all tokens on each cluster (clusters are specified as a query parameter).
```

Comment:
This implementation paragraph would be made clearer and easier to read as a numbered (outline) list of steps.
e.g. 
1. Load all tokens on each cluster...
2. Compute latest version...
3. Perform sync...
  a. Update token description...
  b. Hard-delete..

Reply:
How are you handling this case:

1. Read all tokens.
2. Determine latest.
(client updates a different cluster with a newer version)
3. Go to write the latest on all clusters.
4. Syncer overwrites newer version of token

In short, we need functionality that rejects updates if the existing token data is newer.